packages/cloud-cognitive/src/components/EditTearsheet/EditTearsheetStep.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright IBM Corp. 2021, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { forwardRef, useContext, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { Column, FormGroup, Row } from 'carbon-components-react';
+import { StepsContext, StepNumberContext } from './EditTearsheet';
+import { pkg } from '../../settings';
+import pconsole from '../../global/js/utils/pconsole';
+import { usePreviousValue, useRetrieveStepData } from '../../global/js/hooks';
+
+const componentName = 'EditTearsheetStep';
+const blockClass = `${pkg.prefix}--tearsheet-edit__step`;
+
+// Default values for props
+const defaults = {
+  hasFieldset: true,
+};
+
+export let EditTearsheetStep = forwardRef(
+  (
+    {
+      // The component props, in alphabetical order (for consistency).
+
+      children,
+      className,
+      description,
+      disableSubmit,
+      fieldsetLegendText,
+      hasFieldset = defaults.hasFieldset,
+      introStep,
+      onNext,
+      onMount,
+      secondaryLabel,
+      subtitle,
+      title,
+
+      // Collect any other property values passed in.
+      ...rest
+    },
+    ref
+  ) => {
+    const stepsContext = useContext(StepsContext);
+    const stepNumber = useContext(StepNumberContext);
+    const previousState = usePreviousValue({
+      currentStep: stepsContext?.currentStep,
+    });
+
+    useRetrieveStepData({
+      stepsContext,
+      stepNumber,
+      introStep,
+      secondaryLabel,
+      title,
+    });
+
+    // This useEffect reports back the onNext and onMount values so that they can be used
+    // in the appropriate custom hooks.
+    useEffect(() => {
+      if (
+        stepNumber === stepsContext?.currentStep &&
+        previousState?.currentStep !== stepsContext?.currentStep
+      ) {
+        stepsContext?.setOnNext(stepsContext?.currentStep);
+        stepsContext?.setOnMount(onMount);
+      }
+    }, [onMount, onNext, stepsContext, stepNumber, previousState?.currentStep]);
+
+    // Whenever we are the current step, supply our disableSubmit value to the
+    // steps container context so that it can manage the 'Next' button appropriately.
+    useEffect(() => {
+      if (stepNumber === stepsContext?.currentStep) {
+        stepsContext.setIsDisabled(disableSubmit);
+      }
+    }, [stepsContext, stepNumber, disableSubmit, onNext]);
+
+    return stepsContext ? (
+      <div
+        {
+          // Pass through any other property values as HTML attributes.
+          ...rest
+        }
+        className={cx(blockClass, className, {
```

Comment:
this is a personal preference thing, but i figured i'd point it out. i think it's good practice to avoid having logic or function calls in the HTML portion of react components. i would personally move this out to a variable prior to the return. something like `stepClassName` for instance to clean the HTML more clean.